app/models/asset_field_type.rb

```diff
@@ -52,6 +53,29 @@ def mapping
 
   private
 
+  def is_image(mime_type)
+    ['image/jpeg', 'image/pjpeg', 'image/png','application/pdf' ,'image/x-png', 'image/gif'].include?(mime_type)
+  end
+
+  def format_data(metadata_hash)
+   return metadata_hash unless is_image(metadata_hash[:content_type])
```

Comment:
As a potential solution to Elliott's concerns, couldn't you just remove the `:thumbnail` and `:paperclip_optimizer` strings from the `processors` array in this method?